rumqttd/src/router/routing.rs

```diff
@@ -334,8 +342,14 @@ impl Router {
             code: ConnectReturnCode::Success,
         };
 
+        let properties = ConnAckProperties {
```

Comment:
we are doing this for clients connecting to v4 also, and for them we will panic at `protocol::v4::write` method

Reply:
to replicate:
1. start broker with default settings
2. run `rumqttc/examples/syncpubsub.rs`

Reply:
can we add a simple integration test to verify this behavior in the CI -> 
- broker starts, 
- a client connects to broker, 
- client sends 10 qos 1 messages, 
- if it receives 10 pubacks then test is successful 

all this can happen in one file, like `rumqttd/tests/basic.rs`

Reply:
This is `panic`-y section https://github.com/bytebeamio/rumqtt/blob/tame/rumqttd/src/protocol/v4/mod.rs#L345-L360

Reply:
Hey, good catch! I can think of two options off the top of my head:
1. Ignore the properties in v4 for all packets, e.g.
```diff
- Packet::ConnAck(connack, None)
+ Packet::ConnAck(connack, _)
```
2. Only pass down `ConnAckProperties` if we are connecting to v5. This check is hard to do in router. So I would prefer going with the formal fix.

Let me know if you have some other ideas :100:

Reply:
> 1. Ignore the properties in v4 for all packets, e.g.

This sounds good to me. Are properties only used for mqtt v5?

Reply:
went with option 1. i.e. ignoring connack properties. we don't need to ignore for other packet types as they will be None always.